### PR TITLE
DDF-1914: migration:export command doesn't have a Subject when run

### PR DESCRIPTION
**Pull request by lessarderic**
Moved Subject check code from CatalogMigratableImpl class to ExportCommand.
Updated all unit tests, feature files and pom files accordingly.

Reviewers: @stustison, @pklinef

### CODE REVIEW

File: catalog/core/catalog-core-migratable/pom.xml [hunk 1]
```diff
@@ -152,7 +152,7 @@
                                         <limit>
                                             <counter>BRANCH</counter>
                                             <value>COVEREDRATIO</value>
-                                            <minimum>0.70</minimum>
+                                            <minimum>0.65</minimum>
```
Comment (brendan-hofmann) [inline catalog/core/catalog-core-migratable/pom.xml]: Test coverage went down. :( I assume due to the refactoring, the ratio of critical path to guff has gone down. Could you confirm that we're not missing anything coverage-wise though?

File: platform/migration/platform-migration-commands/src/main/java/org/codice/ddf/migration/commands/ExportCommand.java [hunk 2]
```diff
@@ -76,12 +85,37 @@ protected Object doExecute() {
...
                 outputWarningMessage(String.format(FAILED_EXPORT_MESSAGE, exportDirectory));
             }
         } catch (Exception e) {
             outputErrorMessage(String.format(ERROR_EXPORT_MESSAGE, e.getMessage()));
         }
+
         return null;
     }
 
+    private Subject getSubject() {
```
Comment (brendan-hofmann) [inline platform/migration/platform-migration-commands/src/main/java/org/codice/ddf/migration/commands/ExportCommand.java]: While we're in here refactoring, this method seems like a better candidate for the Security utilities as well. I don't think this is the only command that we want to fall back to the system user, is it?

File: catalog/core/catalog-core-migratable/pom.xml (hunk 1 above)
Comment (lessarderic) [inline catalog/core/catalog-core-migratable/pom.xml]: We have another ticket (DDF-1928) that will rework the unit code and unit tests in the next sprint, so we should be able to bring the coverage back up to at least 75%.

File: platform/migration/platform-migration-commands/src/main/java/org/codice/ddf/migration/commands/ExportCommand.java (hunk 2 above)
Comment (lessarderic) [inline platform/migration/platform-migration-commands/src/main/java/org/codice/ddf/migration/commands/ExportCommand.java]: Good comment. Will be addressed as part of DDF-1928.

**Comment by pklinef** [comment]
:+1:

**Pull request merged**